design/how-does-data-distribution-work.md

```diff
@@ -0,0 +1,83 @@
+
+**Server team (struct TCTeamInfo)**: A server team is a group of k servers that host the same key ranges, where k is the replication factor that is usually three. A server team is healthy if every server in the team is healthy and those servers’ localities satisfies the replication requirement. Servers are grouped into server teams to reduce the possibility of data unavailability events at the event of k server failures. 
+
+**Machine team (struct TCMachineTeamInfo)**: A machine team is a group of k machines, where k is the replication factor. Each server team must be on a machine team, meaning that each server in the server team is on a machine in the machine team and that no two servers are on the same machine. Similar to the purpose of server teams, machine teams are used to reduce the possibility of data unavailability events at the event of k _machine_ failures. A machine team is healthy if every machine on the team is healthy and machines’ localities satisfy the replication policy.
+
+**TeamCollection**: It has a global view of all servers and server teams, machines and machine teams. With the information, it creates server teams and machine teams. It also maintains the configuration setting for DD, which is used to create teams and decide which type of storage servers to recruit.
+
+**Shard (struct DDShardInfo)**: A shard is a key range. A shard is maintained by a server team. A server team is responsible for lots of shards. Each shard has similar amount of data. When a shard has too much data or has too much write traffic, it will be split into multiple shards and redistributed to server teams. Likewise, when a shard has too small data, it can be merged with its neighbors.
+
+**RelocateShard (struct RelocateShard)**: A RelocateShard records the key range that need to be move among servers and the data movement’s priority. DD always move shards with higher priorities first.
+
+**Data distribution queue (struct DDQueueData)**: It receives shards to be relocated (i.e., RelocateShards), decides which shard should be moved to which server team, prioritizes the data movement based on relocate shard’s priority, and control the progress of data movement based on servers’ workload. 
+
+**Special keys in system key space**: DD saves its state in system keyspace to recover from failure and to ensure every process (e.g., proxies, tLogs and storage servers) has a consistent view of which storage server is responsible for which key range. 
+
+*serverKeys* sub-space (\xff/serverKeys/): It records the start key of each shard a server is responsible for. The format is *\xff/serverKeys/[serverID]/[start_key]*. To get start keys of all shards for a server, DD can read the key range with prefix *\xff/serverKeys/[serverID]/*.
```

Comment:
Perhaps also briefly state what the values here are, to avoid confusion with keys/values, same as below